dash-pipeline/SAI/sai_api_gen.py

```diff
@@ -380,6 +380,8 @@ def write_sai_files(sai_api):
                 for table_name in all_table_names:
                     if table_ref.endswith(table_name):
                         param[OBJECT_NAME_TAG] = table_name
+                    if table_name.endswith(table_ref):
```

Comment:
Why doesn't the upper condition work?

Reply:
this is required to generate correct object id for the attribute reference. Without this fix:
```
`260     /**
261      * @brief Action route_vnet parameter VNET_ID
262      *
263      * @type sai_object_id_t
264      * @flags CREATE_AND_SET
265      * @objects SAI_OBJECT_TYPE_ <--------------
266      * @allownull true
267      * @default SAI_NULL_OBJECT_ID
268      * @validonly SAI_OUTBOUND_ROUTING_ENTRY_ATTR_ACTION == SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET
269      */
270     SAI_OUTBOUND_ROUTING_ENTRY_ATTR_VNET_ID,```

Reply:
My question is about upper condition. Why `if table_ref.endswith(table_name)` does not catch that? What is `table_ref` and `table_name` equal to in your case?

Reply:
@marian-pritsak fixed to use the table name for only the user portion as discussed offline. Dont need this change now.

Reply:
@marian-pritsak added the fix for the key names as well. Plz check